src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2808,6 +2865,13 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                     tmp = handle_intrinsic_float(al, args, x.base.base.loc);
                 }
                 return;
+            } else if (call_name == "len") {
+                if (args.size() != 1) {
+                    throw SemanticError("len() takes exactly one argument (" +
+                        std::to_string(args.size()) + " given)", x.base.base.loc);
+                }
```

Comment:
We should put these two checks in `handle_intrinsic_len` IMO. The reason is simple if we are already passing `args` to `handle_intrinsic_len` then why keep a portion of `args` handling outside `handle_intrinsic_len`.